File: internal/repository/mongodb.go
package repository

import (
	"context"
	"fmt"
	"time"

	"github.com/miti99/store-scraper-bot-go/internal/config"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
	"go.uber.org/zap"
)

var (
	client   *mongo.Client
	database *mongo.Database
)

func InitMongoDB(cfg *config.Config) error {
	ctx, cancel := context.WithTimeout(context.Background(), cfg.MongoTimeout)
	defer cancel()

	clientOptions := options.Client().ApplyURI(cfg.MongoURI)

	var err error
	client, err = mongo.Connect(ctx, clientOptions)
	if err != nil {
		return fmt.Errorf("failed to connect to MongoDB: %w", err)
	}

	// Ping to verify connection
	if err := client.Ping(ctx, nil); err != nil {
		return fmt.Errorf("failed to ping MongoDB: %w", err)
	}

	database = client.Database(cfg.MongoDatabase)
	cfg.Logger.Info("Connected to MongoDB",
		zap.String("database", cfg.MongoDatabase),
		zap.String("uri", cfg.MongoURI))

	return nil
}

func Close() error {
	if client != nil {
		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
		defer cancel()
		return client.Disconnect(ctx)
	}
	return nil
}

func GetDatabase() *mongo.Database {
	return database
}

func GetCollection(name string) *mongo.Collection {
	return database.Collection(name)
}

File: src/repository/mongodb.js
import { MongoClient } from 'mongodb';

let client;
let database;

export async function initMongoDB(config) {
  client = new MongoClient(config.mongoUri, {
    serverSelectionTimeoutMS: config.mongoTimeoutMs,
  });
  await client.connect();
  await client.db(config.mongoDatabase).command({ ping: 1 });
  database = client.db(config.mongoDatabase);
  config.logger.info(
    { database: config.mongoDatabase, uri: config.mongoUri },
    'Connected to MongoDB',
  );
}

export async function closeMongoDB() {
  if (client) await client.close();
}

export function getDatabase() {
  if (!database) throw new Error('MongoDB not initialized');
  return database;
}

export function getCollection(name) {
  return getDatabase().collection(name);
}
